R/hausdorffDistPlot.R

```diff
@@ -24,48 +36,113 @@
 #' Hie et al (2019): Geometric sketching compactly summarizes the
 #' single-cell transcriptomic landscape. Cell Systems 8, 483–493.
 #'
+#' Song et al (2022): scSampler: fast diversity-preserving subsampling of
+#' large-scale single-cell transcriptomic data.
+#' bioRxiv doi:10.1101/2022.01.15.476407
+#'
 #' Huttenlocher et al (1993): Comparing images using the Hausdorff
 #' distance. IEEE Transactions on Pattern Analysis and Machine
 #' Intelligence 15(9), 850-863.
 #'
-#' @return
-#' Invisibly, a \code{data.frame} with three columns: the size of the sketch
+#' @return Invisibly, a \code{data.frame} with four columns: the subsampling
+#' method (method), the size of the sketch
 #' as the number of samples (N), the size of the sketch as a fraction
 #' of the original data set (frac), and the robust Hausdorff distance
 #' (HausdorffDist).
 #'
 #' @examples
+#' ## Generate example data matrix
 #' mat <- matrix(rnorm(1000), nrow = 100)
+#'
+#' ## Generate diagnostic Hausdorff distance plot
+#' ## (including all available methods)
 #' hdp <- hausdorffDistPlot(mat, Nvec = c(10, 25, 50))
 #'
+#' ## Provide additional arguments for geosketch
+#' hdp <- hausdorffDistPlot(mat, Nvec = c(10, 25, 50), Nrep = 2,
+#'                          extraArgs = list(geosketch = list(max_iter = 100)))
+#'
 #' @importFrom ggplot2 ggplot aes geom_ribbon geom_point geom_line
 #'     theme_bw scale_x_continuous labs theme element_text
 #' @importFrom dplyr %>% group_by summarize
+#' @importFrom stats runif
 #'
 hausdorffDistPlot <- function(mat, Nvec, Nrep = 5, q = 1e-4,
-                              doPlot = TRUE, ...) {
+                              doPlot = TRUE, seed = NULL,
+                              methods = c("geosketch", "scsampler"),
```

Comment:
Would it be useful to add "unif" or "uniform" as one of the methods? This could give a frame of reference as to how much better these are compared to naive sampling.

Reply:
Yes, great idea! (I think I would prefer to call it "uniform").

Reply:
I have added this in 9e97f6e